### ISSUE TITLE
[FEATURE] Add a format flag to allow parsing with , as separator

### ISSUE THREAD

**Issue by KaiserKarel** [Alexhuszagh/rust-lexical#58]
## Problem
Dutch floats are formatted like so: `101.123,456`, where the `.` is the separator, and the comma is used for the fraction.

Afaik, it is not possible to add a format flag to allow parsing dutch floats.

Some way to configure the parser to allow that would be great.

**Comment by myrrlyn** [comment]
I am amenable to this. Currently it appears that we implicitly assume `,` is a valid digit separator and `.` is the fractional separator. Am I correct to assume that the only change necessary to support this locale is swapping `.` and `,`?

**Comment by Alexhuszagh** [comment]
I believe so @myrrlyn, as I believe this is the same for French and Belgian numbers. The only issue is this does require a fair amount of removing hard-coded assumptions about the decimal place, which should be decently trivial. I believe this is the only place where this assumption is currently hard-coded.

https://github.com/Alexhuszagh/rust-lexical/blob/master/lexical-core/src/atof/algorithm/format/traits.rs#L260

**Comment by Alexhuszagh** [comment]
This should be fairly trivial, on further glance. I'll push it after I finish a major bug release tomorrow. The only primary issues are:

1. The hard-coded assumption of the exponent type.
2. An addition to the bitflags.

We currently have more than enough bitflags to cover this case, and the digit separator is already easily handled.

We currently have 64-bit flags for the formats, and the lower 56 are reserved for format flags, and the top 8 are for an 8-bit (in reality, only ASCII) digit separator.

```rust
    #[inline]
    const fn digit_separator_to_flags(ch: u8) -> u64 {
        (ch as u64) << 56
    }
```

Currently our highest digit separator is:

```rust
            #[doc(hidden)]
            const SPECIAL_DIGIT_SEPARATOR               = 0b0000000000000000000100000000000000000000000000000000000000000000;
```

Which gives us about 11 more flags to use. This is an excellent use of one of those.

**Comment by Alexhuszagh** [comment]
Adding on to this issue: we should also allow formats with '^' as a valid exponent separator, even for radices <= 0xE, which would build towards to slow removal of #45.

**Comment by Alexhuszagh** [comment]
@KaiserKarel I've added tentative support for this on master (to be released with the 0.8 branch). Please ignore the broken test suite: I had to rework a lot of internal functions in order to accommodate this, so the test suite is expected to be broken just due to function signatures alone.

The general approach would be something as follows:

```rust
extern crate lexical_core;
// All besides `unwrap` is a `const fn` for both builders.
let format = lexical_core::NumberFormat::builder().exponent_default(b',').build().unwrap();
let options = lexical_core::ParseFloatOptions::builder().format(format).build().unwrap();

// Ok(5.0)
let result = lexical_core::parse_with_options(b"5,0", &options);
```

This does not require `feature = "format"`, which mitigates any performance penalties of this.

**Comment by Alexhuszagh** [comment]
Implemented as of [lexical v6.0.0](https://crates.io/crates/lexical/6.0.0) and [lexical-core v0.8.0](https://crates.io/crates/lexical-core/0.8.0).

A code sample supporting this is as follows:

```rust
 // This creates a format to parse a European-style float number.
 // The decimal point is a comma, and the digit separators (optional)
 // are periods.
 const EUROPEAN: u128 = lexical_core::NumberFormatBuilder::new()
     .digit_separator(b'.')
     .build();
 let options = lexical_core::ParseFloatOptions::builder()
     .decimal_point(b',')
     .build()
     .unwrap();
 assert_eq!(
     lexical_core::parse_with_options::<f32, EUROPEAN>(b"300,10", &options),
     Ok(300.10)
 );
```
